nomad/resource_job.go

```diff
@@ -56,6 +56,11 @@ func resourceJobRegister(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error parsing jobspec: %s", err)
 	}
 
+	// Inject the Vault token if provided
+	if providerConfig.vaultToken != nil {
```

Comment:
This check isn't really necessary, I don't think; setting the VaultToken to nil would have the same effect as not setting it at all.

Reply:
I was sure I had issues with that but it seems to be working without the condition now